pytestqt/wait_signal.py

```diff
@@ -138,6 +213,63 @@ def _cleanup(self):
             _silent_disconnect(signal, self._quit_loop_by_signal)
         self._signals = []
 
+    def get_params_as_str(self):
+        if not self.all_args:
+            return ""
+
+        if len(self.all_args[0]) == 1:
+            # we have a list of tuples with 1 element each (i.e. the signal has 1 parameter), it doesn't make sense
+            # to return something like "[(someParam,), (someParam,)]", it's just ugly. Instead return something like
+            # "[someParam, someParam]"
+            args_list = [arg[0] for arg in self.all_args]
+        else:
+            args_list = self.all_args
+
+        return str(args_list)
+
+    def get_timeout_error_message(self):
+        if self.check_params_callback is not None:
+            return "Signal {signal_name} emitted with parameters {params} " \
+                   "within {timeout} ms, but did not satisfy " \
+                   "the {cb_name} callback".format(signal_name=self.signal_name, params=self.get_params_as_str(),
+                                                   timeout=self.timeout,
+                                                   cb_name=self.get_callback_name(self.check_params_callback))
+        else:
+            return "Signal {signal_name} not emitted after {timeout} ms".format(signal_name=self.signal_name,
+                                                                               timeout=self.timeout)
+
+
+SignalAndArgs = namedtuple("SignalAndArgs", ["signal_name", "args"])
+
+
+def _get_readable_signal_with_optional_args(self):
+    if self.args:
+        args_as_string = []
+        for arg in self.args:
+            if type(arg) is str:
+                args_as_string.append("'" + str(arg) + "'")
+            else:
+                args_as_string.append(str(arg))
+        args_as_list_string = ", ".join(args_as_string) if len(args_as_string) > 1 else args_as_string[0]
+        args = "({})".format(args_as_list_string)
+    else:
+        args = ""
+
+    # remove signal parameter signature, e.g. turn "some_signal(QString,int)" to "some_signal", because we're adding
+    # the actual parameters anyways
+    signal_name = self.signal_name
+    if '(' in signal_name:
+        signal_name = signal_name[:signal_name.index('(')]
+
+    return signal_name + args
+
+
+SignalAndArgs.__str__ = _get_readable_signal_with_optional_args
+SignalAndArgs.__repr__ = _get_readable_signal_with_optional_args
```

Comment:
Might make more sense to have a custom class instead of abusing a `namedtuple` and then monkeypatching stuff on it?